Apps/Contoso.WPF.Puppet/MainWindow.xaml.cs

```diff
@@ -84,15 +91,112 @@ private void trackEvent_Click(object sender, RoutedEventArgs e)
             var propertiesDictionary = Properties.Where(property => property.Key != null && property.Value != null)
-                                                 .ToDictionary(property => property.Key, property => property.Value);
+                .ToDictionary(property => property.Key, property => property.Value);
             Analytics.TrackEvent(name, propertiesDictionary);
         }
 
+        #region Crash
+
+        public class NonSerializableException : Exception
+        {
+        }
+
+        private void CrashWithTestException_Click(object sender, RoutedEventArgs e)
+        {
+            Crashes.GenerateTestCrash();
+        }
+
+        private void CrashWithNonSerializableException_Click(object sender, RoutedEventArgs e)
+        {
+            throw new NonSerializableException();
+        }
+
+        private void CrashWithDivisionByZero_Click(object sender, RoutedEventArgs e)
+        {
+            _ = 42 / int.Parse("0");
+        }
+
+        private void CrashWithAggregateException_Click(object sender, RoutedEventArgs e)
+        {
+            throw GenerateAggregateException();
+        }
+
+        private static Exception GenerateAggregateException()
+        {
+            try
+            {
+                throw new AggregateException(SendHttp(), new ArgumentException("Invalid parameter", ValidateLength()));
+            }
+            catch (Exception e)
+            {
+                return e;
+            }
+        }
+
+        private static Exception SendHttp()
+        {
+            try
+            {
+                throw new IOException("Network down");
+            }
+            catch (Exception e)
+            {
+                return e;
+            }
+        }
+
+        private static Exception ValidateLength()
+        {
+            try
+            {
+                throw new ArgumentOutOfRangeException(null, "It's over 9000!");
+            }
+            catch (Exception e)
+            {
+                return e;
+            }
+        }
+
         private void CrashWithNullReference_Click(object sender, RoutedEventArgs e)
         {
             string[] values = { "a", null, "c" };
             var b = values[1].Trim();
             System.Diagnostics.Debug.WriteLine(b);
         }
+
+        private async void CrashInsideAsyncTask_Click(object sender, RoutedEventArgs e)
+        {
+            await FakeService.DoStuffInBackground();
+        }
+
+        private static class FakeService
+        {
+            public static async Task DoStuffInBackground()
+            {
+                await Task.Run(() => throw new IOException("Server did not respond"));
+            }
+        }
+
+        private void crashesEnabled_Checked(object sender, RoutedEventArgs e)
+        {
+            if (crashesEnabled.IsChecked.HasValue)
+            {
+                Crashes.SetEnabledAsync(crashesEnabled.IsChecked.Value).Wait();
+            }
+        }
+
+        public int val
```

Comment:
Naming

Reply:
I didn't write this code, I wonder how it ended up in the PR, will look.